_pages/mobile-interview-project.md

```diff
@@ -8,40 +8,55 @@ header:
+
+This site outlines the info and guidelines you’ll need to complete Square’s Mobile Take-Home Project: an employee directory app.
+
+You’ll have a week to complete the question – the actual engineering work should take about 3-4 hours, and you can split this across the week however you see fit (all at once, over a few days, etc). If you have any questions during the process, please share them with your recruiting contact and we’ll get back to you ASAP.
+
+Good luck!
 
 # Overview
-Build an employee directory app that shows a list of employees parsed from the provided JSON endpoint.
 
-Take care to properly handle errors returned from the endpoint (or other network errors like timeouts), and ensure you do not waste network bandwidth – load expensive resources such as large photos on-demand only. Photos at a given URL will never change. Once one is loaded, you do not need to reload the photo. If an employee’s photo changes, they will be given a new photo URL (eg, you can treat the URL as a GUID).
+Build an employee directory app that shows a list of employees from the provided endpoint.
+
+The app should display a list (or any kind of collection view!) which shows all the employees returned from the JSON endpoint described below. Each item in the view should contain a summary of the employee, including their photo, name, and team at minimum. You may add more information to the summary if you want, or sort employees in any fashion you’d like – sort and group by name, team, etc.
 
-All screens and views which load from the network should display proper loading, empty, and error states when content is not available. If images fail to load, displaying a placeholder is fine.
+There should be some UX to reload the employee list from within the app at any time. The UX can be done in any way you want: a button, pull-to-refresh, etc.
 
-The employee list ***does not*** need to be persisted to disk – you can reload it from the network on each app launch. On the contrary, images ***should*** be cached on disk as to not waste device bandwidth.
+If there is any additional UI/UX you would like to add, feel free to do so! We only ask that you please do not build any more screens than this list. Do not worry about building custom controls or UI elements – using system-provided, standard elements is totally fine.
 
-You should architect the app in such a way that you can provide unit tests for critical pieces of functionality. There is no minimum or maximum amount of test coverage we’re looking for – instead try to provide enough coverage such that regressions are caught by automated tests. Only worry about unit tests – you can skip view snapshot or integration tests.
+Be sure to appropriately handle the normal variety of errors when querying an endpoint. The app should display useful loading, empty, and error states where appropriate. If images fail to load, displaying a placeholder is fine.
```

Comment:
This opens the door to UI tests Espresso or XCUnit, which require an emulator because we're no longer saying "only worry about unit tests"?  At least for Android I think it might be helpful then to say something like.

```
For Android, if you write UI Tests (i.e. Espresso Tests) be sure to indicate which emulator you used for testing. For example: Pixel 4XL running API 30.
```